packages/teams-js/src/public/dialog.ts

```diff
@@ -290,4 +242,227 @@ export namespace dialog {
     };
     return dialogInfo;
   }
+
+  /**
+   * @hidden
+   * Hide from docs
+   * --------
+   * Convert AdaptiveCardDialogInfo to DialogInfo to send the information to host in {@linkcode adaptiveCard.open} API.
+   *
+   * @internal
+   */
+  export function getDialogInfoFromBotAdaptiveCardDialogInfo(
+    botAdaptiveCardDialogInfo: BotAdaptiveCardDialogInfo,
+  ): DialogInfo {
+    const dialogInfo: DialogInfo = {
+      card: botAdaptiveCardDialogInfo.card,
+      height: botAdaptiveCardDialogInfo.size ? botAdaptiveCardDialogInfo.size.height : DialogDimension.Small,
+      width: botAdaptiveCardDialogInfo.size ? botAdaptiveCardDialogInfo.size.width : DialogDimension.Small,
+      title: botAdaptiveCardDialogInfo.title,
+      completionBotId: botAdaptiveCardDialogInfo.completionBotId,
+    };
+    return dialogInfo;
+  }
+
+  export namespace adaptiveCard {
+    /**
+     * Allows app to open an adaptive card based dialog.
+     *
+     * @remarks
+     * This function cannot be called from inside of a dialog
+     *
+     * @param adaptiveCardDialogInfo - An object containing the parameters of the dialog module.
+     * @param submitHandler - Handler that triggers when a dialog calls the {@linkcode submit} function or when the user closes the dialog.
+     * @param messageFromChildHandler - Handler that triggers if dialog sends a message to the app.
+     *
+     * @returns a function that can be used to send messages to the dialog.
+     */
+    export function open(
+      adaptiveCardDialogInfo: AdaptiveCardDialogInfo,
+      submitHandler?: DialogSubmitHandler,
+      messageFromChildHandler?: PostMessageChannel,
```

Comment:
For AC-based dialogs, AFAIU, no user-provided code runs. It's all data driven. How is the dialog supposed to send messages back to the launch page and/or respond to messages sent from the parent?

Reply:
This is a good question, let me poke around a little (and possibly talk with teams about it if the documentation isn't clear). I suspect you're right, which would be great

Reply:
Got confirmation from teams: shouldn't be any reason to keep this for AC dialogs

Reply:
Updated this: removed the handler from here and moved all of the message handling code down into dialog.url